leges.go

```diff
@@ -76,116 +33,142 @@ func (e *ErrExprRunFailed) Error() string {
 }
 
 type ErrExprCompileFailed struct {
-	Env     Attributes
-	Policy  Policy
-	Request Request
-	Err     error
+	Env    Attributes
+	Policy Policy
+	Err    error
 }
 
 func (e *ErrExprCompileFailed) Unwrap() error {
 	return e.Err
 }
 
 func (e *ErrExprCompileFailed) Error() string {
-	return "failed to compile expression"
+	return "failed to compileCondition expression"
 }
 
-// Leges is the struct that contains the policies and can be used to requests
-type Leges struct {
-	policies []Policy
+// statute is one policy and the compiled program of the condition
+type statute struct {
```

Comment:
"statute" lol nice!

as a part-time law nerd lol can't help but point that a statute is a law that is passed by a legislative body (eg an act of parliament). the idea would be that statutes are above other common laws and have a higher priority.

but that doesn't translate well here, plus introducing a new name/concept makes it a bit harder to understand what's going on.

imo just call it a stupid name like policyProgram or internalPolicy or cachedPolicy or something else that says it contains a policy along with its cached program.

Reply:
agree, much better, I opt for cachedPolicy, I don't like having the program notion in Leges, but I couldn't find any way around it.